exercises/lamar_moore/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,113 @@
+#include <iostream>
+#include <vector>
+#include <memory>
+#include <stdlib.h>
+#include "Shape.h"
+#include "Square.h"
+#include "Rectangle.h"
+#include "Circle.h"
+#include "Triangle.h"
+#include "ShapeSorter.h"
+
+using namespace std;
+using namespace Shapes;
+
+namespace
+{
+	/** Obtains a random number between 0 and 3 which corresponds to ShapesTypes.
+	*/
+	ShapeType getRandomShape()
```

Comment:
Nice way of making any number of shapes.

**Something to consider**
When you're developing in Mantid. We would be using UnitTests CXXTest rather than using hand-rolled main functions for testing. In that case It's nice to see explicit creation rather than the use of random. It also helps a lot with debugging when you want to trace the failure of a particular test case.